include/revng-c/mlir/Pipes/MLIRContainer.h

```diff
@@ -0,0 +1,65 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "llvm/ADT/DenseMap.h"
+
+#include "mlir/IR/BuiltinOps.h"
+#include "mlir/IR/FunctionInterfaces.h"
+#include "mlir/IR/MLIRContext.h"
+
+#include "revng/Pipeline/Container.h"
+
+#include "revng-c/Pipes/Kinds.h"
+
+namespace revng::pipes {
+
+class MLIRContainer : public pipeline::Container<MLIRContainer> {
+public:
+  static const char ID;
+  static constexpr auto Name = "mlir-module";
+  static constexpr auto MIMEType = "application/x.mlir.bc";
+
+private:
+  // unique_ptr is used to allow moving the context.
+  std::unique_ptr<mlir::MLIRContext> Context;
+  mlir::OwningOpRef<mlir::ModuleOp> Module;
+
+public:
+  explicit MLIRContainer(const llvm::StringRef Name) :
+    pipeline::Container<MLIRContainer>(Name) {
+    clear();
+  }
+
+  mlir::MLIRContext *getContext() { return Context.get(); }
+  mlir::ModuleOp getModule() { return *Module; }
+  void setModule(mlir::OwningOpRef<mlir::ModuleOp> &&NewModule);
```

Comment:
I'm not super stoked about this method. What do you think about creating the module and the context in the constructor, and having users populated that one instead of creating their own and then `setModule` it?

Reply:
Usually that would be fine, but the import from LLVM returns a new module which where this function is needed.

Reply:
Ah OK, so it's due to a "limitation" of some API we rely on. Gotcha.